src/plugins/intel_gna/src/transformations/matops_to_dwsc.cpp

```diff
@@ -0,0 +1,261 @@
+    dwsc_weights_const->set_friendly_name("dwsc_weights");
+
+    // Initializing group convolution parameters
+    const Strides& strides = Strides({1, 1});
+    const CoordinateDiff& pads_begin = CoordinateDiff({0, 0});
+    const CoordinateDiff& pads_end = CoordinateDiff({0, 0});
+    const Strides& dilations = Strides({1, 1});
+    const ngraph::op::PadType& auto_pad = ngraph::op::PadType::EXPLICIT;
+
+    // Initializing the dwsc node
+    std::shared_ptr<ov::op::v1::GroupConvolution> new_dwsc;
+    if (4 != input_shape.size()) {
+        std::shared_ptr<ov::op::v1::Reshape> input_4d;
+        if (W > 1) {
+            auto n_elements = input_shape[0] * input_shape[1];
+            if (n_elements < 8 || n_elements > 65528 || n_elements % 8 != 0)
+                return false;
+            auto new_transpose = std::make_shared<ngraph::op::Transpose>(
+                math_node->input_value(0),
+                ngraph::op::Constant::create(element::Type_t::i64, Shape{2}, {1, 0}));
+            input_4d = std::make_shared<ngraph::opset1::Reshape>(
+                new_transpose->output(0),
+                ngraph::op::Constant::create(ngraph::element::i64, Shape{4}, {N, C, H, W})->output(0),
+                false);
+        } else
+            input_4d = std::make_shared<ngraph::opset1::Reshape>(
+                math_node->input_value(0),
+                ngraph::op::Constant::create(ngraph::element::i64, Shape{4}, {N, C, H, W})->output(0),
+                false);
+        new_dwsc = std::make_shared<ngraph::opset1::GroupConvolution>(input_4d->output(0),
+                                                                      dwsc_weights_const->output(0),
+                                                                      strides,
+                                                                      pads_begin,
+                                                                      pads_end,
+                                                                      dilations,
+                                                                      auto_pad);
+    } else {
+        new_dwsc = std::make_shared<ngraph::opset1::GroupConvolution>(math_node->input_value(0),
+                                                                      dwsc_weights_const->output(0),
+                                                                      strides,
+                                                                      pads_begin,
+                                                                      pads_end,
+                                                                      dilations,
+                                                                      auto_pad);
+    }
+
+    new_dwsc->set_friendly_name("replace_math_operation");
+
+    std::shared_ptr<ov::Node> skip_node;
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::Multiply>(math_node)) {
+        // creating a bias node
```

Comment:
if we move content of this if statement to separate method with valuable name if it would be self explanatory.

Reply:
Done